src/contexts/formContext.tsx

```diff
@@ -0,0 +1,33 @@
+import { createContext, useCallback, useContext, useMemo, useReducer } from 'react';
+import { CustomPokemon, FormActionType, FormContextProps } from 'types';
+import { formReducer, initialFormState } from 'reducers';
+
+export const FormContext = createContext<FormContextProps>({} as FormContextProps);
+
+export const FormContextProvider = ({ children }: { children: JSX.Element }) => {
+  const [formState, dispatch] = useReducer(formReducer, initialFormState);
```

Comment:
https://github.com/redux-form/redux-form#%EF%B8%8F-attention-%EF%B8%8F

As you use useContext + useReducer combination - this works for you too.